design/FY2019/Refactor and cache two psychrometric functions.md

```diff
@@ -0,0 +1,77 @@
+| EP_cache_PsyTsatFnPb  | 4.30%                                              | 2                              | 0.47%                                      | 1                              | 0.12%                                          | 1                              | N/A                                          | 0                              | 0.72                                         | 2                              |
+| EP_cache_PsyTsatFnHPb | 4.20%                                              | 3                              | 0.31%                                      | 1                              | 0.04%                                          | 2                              | N/A                                          | 0                              | 0.60%                                        | 2                              |
+
+We further took some example files and analyzed where the big diffs show in each test case. The big deviations mainly appear in fields including heating and cooling capacity, convergence information, annual peak (minimum and maximum) values and setpoint not met (Table 3). This indicates that at some extreme conditions, minor (10^-16) precision difference in the calculation of psychrometric function can cause cumulative effect in simulation results.
+
+We summarized the diff analysis of the ESO file with each malfunctioning test case for the PsyTsatFnPb function, including (1) if the file has large abs value at large rel diff hours for some attributes, (2) maximum #hours with big ref diff at abs values out of 8760, and (3) most significant deviated attribute. A more detailed diff analysis by attributes is attached in a spreadsheet. For most of the files with large math diffs, the diff happens only in a few hours for some sensitive fields, and by the diff-of-warning-messages many of them are triggered by the inconsistent number of iterations for convergence. For example, the reference supermarket has large math diff in total heating energy during one peak hour only. For 5ZoneVAV-ChilledWaterStorage-Mixed, the cached version somehow converges in less number of iterations for some HVAC loops, and that's accounting for the diffs in Solver Iteration Count. Other files with diffs originally have some existing severe error and warnings which I assume can be vulnerable now, and their behaviors can deviate in subroutines such as cooling tower and heat pumps. None of those behaviors leads to large diffs in annual energy use though.
+
+
+**Table 3. Regression diffs analysis of the PsyTsatFnPb function**
```

Comment:
Table 3 got messed up at some point, it's showing up like this when I try to view the rendered version:

![image](https://user-images.githubusercontent.com/849824/64907737-07871600-d6bc-11e9-8a1d-1954fd41c598.png)